### PR TITLE
chore(core): added function to convert large numbers into short form

### PR DESCRIPTION
**Pull request by rohit1290**
Fix https://github.com/Elgg/Elgg/issues/12020

### CODE REVIEW

**Comment by jdalsem** [comment]
would it be an idea to always format numbers in badges? that way we can do this in the badge output automatically (do the check in /navigation/menu/elements/item)

```
$badge = $item->badge;
if (is_numeric($badge)) {
$item->badge = format_number_function($badge);
}
```

maybe also move the function in \Elgg\Values as a new static function

File: engine/lib/output.php [hunk 1]
```diff
@@ -464,3 +464,40 @@ function _elgg_sane_validate_url($url) {
...
+ * Example: elgg_number_format_short(7201); // Output: 7K
+ * Example: elgg_number_format_short(7201,1); // Output: 7.2K
+ *
+ * @param $n         input number to format
+ * @param $precision number of digits in decimal place (default = 0)
+ * @return string
+ *
+ * @since 3.1
+ */
+function elgg_number_format_short($n, $precision = 0) {
```
Comment (jeabakker) [inline engine/lib/output.php]: almost a  i envisioned it, have a look at 
https://github.com/ColdTrick/entity_view_counter/blob/51f8164903e47667584bd94a47258e5169692c49/start.php#L102-L142

**Comment by jdalsem** [comment]
my code is probably not working... just pseudo code

**Comment by jdalsem** [comment]
also... the commit type should be `feat`

File: engine/lib/output.php (hunk 1 above)
Comment (jeabakker) [reply engine/lib/output.php]: don't copy the code ;) 

Thinks I think are better in our code
- language key for suffix
- less complexitiy in if statements `$n > 1000 ` is not noeeded

File: views/default/navigation/menu/elements/item.php [hunk 2]
```diff
@@ -56,6 +56,11 @@
 	$item_vars['class'][] = "elgg-state-selected";
 }
 
+$badge = $item->badge;
```
Comment (jdalsem) [inline views/default/navigation/menu/elements/item.php]: did you test this? maybe it needs to be $item->getData('badge') i am not sure... the tests are probably failing on this

Comment (rohit1290) [reply views/default/navigation/menu/elements/item.php]: I am planning to do this..

```
if (is_numeric($item->badge)) {
$item->badge = format_number_function($item->badge);
}
```
I don't know why but phpunit test does not work on my localhost.

Comment (rohit1290) [reply views/default/navigation/menu/elements/item.php]: `$item->badge` is working fine, I have tested the same in search feature.

**Comment by jdalsem** [comment]
This is looking good. When adding new API functions we also need tests. Could you add a test in phpunit/unit/Elgg/ValuesUnitTest.php

Something like:
```php

	/**
	 * @dataProvider shortNumberProvider
	 */
	public function testCanShortenNumber($number, $precision, $expected) {

		$this->assertEquals(Values::shortFormatOutput($number), $expected);

	}

	public function shortNumberProvider() {
		return [
			[1, 1, 1],
			[1200, 1, '1.2K'],
			[1230, 2, '1.23K'],
			['a', 1, 'a'],
			[1100000000, 1, '1.1M'],
		];
	}
```

not tested yet

**Comment by jdalsem** [comment]
very nice!

**Pull request merged**